### PR TITLE
Feature/members

### PR DESCRIPTION
**Pull request by RailonA**
A rails Members Only project
> This is part of the Forms Project in The Odin Project’s Ruby on Rails Curriculum. Find it at https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication

Set up a User account where you will be able to post only when logged in.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,69 @@
...
+#### Test all in the browser
+- Run `$ rails server`
+- In the browser go to http://127.0.0.1:3000/users/sign_up
+- This page should display :arrow_down:
+- ![screenshot](./app/assets/images/members-only.png)
+- Create a new user by putting informations inside the text fields
+#### Test the tables
+- First run `rails c`
+- In the rails console make sure all the tables are working correctly. Run `User`
+- Then look at all User to make sure that information used to greate the user account was pushed successfully to the database.
```
Comment (enaburekhan) [inline README.md]: Kindly correct spelling error `greate` to `create`

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -5,12 +5,38 @@
...
+	   <div class="notification is-primary global-notification">
+	   <p class="notice"><%= notice %></p>
+	   </div>
+	  <% end %>
+	
+	  <% if flash[:alert] %>
+	   <div class="notification is-primary global-notification">
+	   <p class="alert"><%= alert %></p>
+	   </div>
+	  <% end %>
```
Comment (enaburekhan) [inline app/views/layouts/application.html.erb]: There should be absolutely NO logic in views. Views are only for displaying information that became available to it from its corresponding controller not generating it. Kindly move them to a helper

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<section class="section">
+  <div class="container">
+      <div class="columns">
+          <% if user_signed_in? %>
+          <%=render "profile" %>
+          <% end %>
+          <%= render "feed" %>
```
Comment (enaburekhan) [inline app/views/posts/index.html.erb]: Move this logic to a helper.

File: app/views/shared/_navbar.html.erb [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+              <a class="nav-link">
+                  <%= link_to 'Sign In', new_user_session_path, class: "text-white" %>
+              </a>
+            </li>
+           <li class="nav-item">
+              <a class="nav-link">
+                <%= link_to 'Sign Up', new_user_registration_path, class: "text-white" %>   
+              </a>
+            </li>
+             <% end %>
```
Comment (enaburekhan) [inline app/views/shared/_navbar.html.erb]: Move this logic to a helper.

**Review by enaburekhan — CHANGES_REQUESTED** [review]
# Status: Changes required:full_moon:
**Hi team!
Great job so far on this project:clap::clap:.**

**Major highlights**

- The readme and PR is very descriptive and hence, professional
- The UI is awesome.
- The codebase is clean and codes are clearly written,

However, we still need to make a few changes to get this project approved:muscle:.

**To improve**

- [x] make the develop the default branch. Instead of the master which will always be the default if not changed.

- [x] PostsController

- it uses `before_action` to restrict access to the `:new` and `:create` methods (uses ApplicationController helper method) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

`before_action :signed_in_only!, only: [:new, :create]`
```ruby
def signed_in_only!
  # current_user in ApplicationController must be defined as helper method to make it work
  redirect_to  new_user_session_path unless current_user

  ## alternative solution:
  # if signed_in? is defined in ApplicationController (see stretch requirements)
  redirect_to new_user_session_path unless signed_in?
end
``` 
You can achieve this by using a helper. And then use the before action to restrict access to only `new` and `create` inside the posts controller.
 

- new

1. it initializes an empty post `@post = Post.new` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/posts_new_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))

- [x] As already highlighted below. There should be absolutely NO logic in views. Views are only for displaying information that became available to it from its corresponding controller not generating it. Move all the logic highlighted below in the PR to a helper. Check this [article](http://codefol.io/posts/Where-Do-I-Put-My-Code/) and this [article](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers) on how to use a rails helper. See [screen-shot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from pathwright

- [x] Delete edit and show action in posts_controller.rb. Since they were not used and hence, not required.

Make the required changes and when you are ready, request another review.
Have any question/comment? kindly drop them :point_down: and ping me on slack @ Eric Enaburekhan to respond
Happy hacking:blush:
Eric Enaburekhan

File: app/views/posts/_feed.html.erb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+       <div class="image is-64x64">
+            <figure class="media-left">
+						<%= gravatar_image_tag(post.user.email, size: 64, alt: post.user.name) %>
+            </figure>
+        </div>
+    <%end%>
+        <div class="media-content">
+            <div class="content">
+                <% if user_signed_in? %>
+                    <strong><%= post.user.name %></strong><br/>
```
Comment (enaburekhan) [inline app/views/posts/_feed.html.erb]: Move this logic to a helper. As highlighted in the last review. There should be absolutely no logic in view.

File: app/views/posts/_feed.html.erb [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+                        <span class="icon"><i class="fa fa-link" aria-hidden="true"></i></span>
+                    <% end %>
+                    <%= link_to edit_post_path(post), class: "level-item" do %>
+                        <span class="icon"><i class="fa fa-pencil" aria-hidden="true"></i></span>
+                    <% end %>
+                    <%= link_to post, method: :delete, data: { confirm: "Are you sure you want to delete this post?"} do %>
+                        <span class="icon"><i class="fa fa-trash-o" aria-hidden="true"></i></span>
+                    <% end %>
+                </div>
+                  <%end%>
```
Comment (enaburekhan) [inline app/views/posts/_feed.html.erb]: Move the logic to a helper.

File: app/views/shared/_navbar.html.erb [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
```
Comment (enaburekhan) [inline app/views/shared/_navbar.html.erb]: :point_up:

**Review by enaburekhan — CHANGES_REQUESTED** [review]
# Status: Changes required:full_moon:
**Hi team!
Great job so far. Thank you for implementing some of the changes earlier requested:clap::clap:. We are almost there now. The readme is very descriptive and the codebase is clean, easy to read and understand.
However, we just need to make a few changes to get this project approved:muscle:.**

**To improve**
- [x]  Check my comments below on the view section. As earlier highlighted. There should be absolutely NO logic in views. Views are only for displaying information that became available to it from its corresponding controller not generating it. We still have logics in _feed.html.erb under posts, and _navbar.html.erb. Kindly move them to a helper. 

- [x] Make the develop or development branch the default branch and not the master branch ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/git_flow_requirement.png) from Pathwright)

Make the above-required changes and when you are ready, request another review.
Do have questions/comments? kindly drop them :point_down: and ping me on slack @ Eric Enaburekhan to respond.
Happy hacking:blush:
Eric Enaburekhan.

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
## **Changes Required 🛑♻**

<hr>

Hello there :)
Great job you've done on this app and on the requested changes. You've refactored all the logic from the views, and thanks for the extra work you made on editing and deleting posts. However, there is still one small change in order to get the approval 👍 

- [x] When I create posts with a user, then login with another user, the posts' author becomes the new user, not the user that actually created those posts. Kindly fix this issue.

<hr>

Good luck and happy coding 😊👨‍💻💻

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
## **Changes Required 🛑♻**

<hr>

Hello there :)
Great job you've done on this app and on the requested changes. You've refactored all the logic from the views, and thanks for the extra work you made on editing and deleting posts. However, there is still one small change in order to get the approval 👍 

- [x] When I create posts with a user, then login with another user, the posts' author becomes the new user, not the user that actually created those posts. Kindly fix this issue.

<hr>

Good luck and happy coding 😊👨‍💻💻

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello Ismael Ngalula Antonio & Railon Acosta** 
Good job with the improvements for this project :muscle:  :muscle:
It meets most of the requirements :clap: :clap: :clap:  
There are a couple of details to improve before you can go ahead:

- [x] Don't overuse rubocop disable
- [x] Fix rubocop offenses
 
Please request another review once you are ready! :+1: 
 
Happy coding :relaxed:
 
Oscar Andrés Russi Porras

**Review by jubaan — APPROVED** [review]

**Pull request merged**